src/k8s-engine/kim.ts

```diff
@@ -0,0 +1,150 @@
+const { EventEmitter } = require('events');
+const process = require('process');
+const { spawn } = require('child_process');
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const paths = require('xdg-app-paths')({ name: 'rancher-desktop' });
+const resources = require('../resources');
+
+const REFRESH_INTERVAL = 5 * 1000;
+
+interface childResultType {
+  stdout: string,
+  stderr: string,
+  code: number
+}
+
+interface imageType {
+  imageName: string,
+  tag: string,
+  imageID: string,
+  size: string,
+}
+
+export default class Kim extends EventEmitter {
+  constructor() {
+    super();
+    this.showedStderr = false;
+    this.refreshInterval = null;
+    this.currentCommand = null;
+  }
+
+  start() {
+    this.stop();
+    this.refreshInterval = setInterval(this.refreshImages.bind(this), REFRESH_INTERVAL);
+  }
+
+  stop() {
+    if (this.refreshInterval) {
+      clearInterval(this.refreshInterval);
+      this.refreshInterval = null;
+    }
+  }
+
+  async runCommand(args: string[], sendNotifications = true): Promise<childResultType> {
+    const child = spawn(resources.executable('kim'), args);
+    const result :childResultType = {
```

Comment:
I think this file needs to be formatted? Probably means something's off in our linter configuration…

Reply:
Anything specific?

Reply:
Sorry should have included this in the comments:
```suggestion
    const result: childResultType = {
```

Reply:
ok